tests/test_template.py

```diff
@@ -38,3 +39,16 @@ def test_find_shapes_from_one_slide(template):
 def test_find_shapes_index_out_of_range(template):
     with pytest.raises(IndexError):
         shapes = template._find_shapes(0, 'logo')
+
+def test_get_all_shapes(template):
+    shapes = template._get_all_shapes()
+    for shape in shapes:
+        assert str(shape.text) and str(shape.name)
+
+def test_copy_tags_to_name(template):
+    template._copy_tags_to_name()
+    all_shapes = template._get_all_shapes()
+    regex_tag = re.compile(r'\{[\s\w]*\}')
+    for shape in all_shapes:
+        if regex_tag.match(shape.text):
+            assert shape.text == shape.name
```

Comment:
Is this really what you wan't to test? Taking a look at the regex you defined in the implementation, for a shape having the text '  {location}  ', you would set the name as '{location}'. Hence, in this case, `shape.text != shape.name`.

Reply:
You're right, this test needs to be fixed :)